src/components/Dashboard/Dashboard.js

```
@@ -49,20 +57,21 @@ class Dashboard extends Component {
       });
   }
 
-  deletePlugin(pluginId) {
+  async deletePlugin(pluginId) {
     const { store } = this.props;
     const storeURL = process.env.REACT_APP_STORE_URL;
     const auth = { token: store.get('authToken') };
     const client = new Client(storeURL, auth);
 
     let response;
     try {
-      response = client.getPlugin(pluginId).then(plugin => plugin.delete());
-      response.then(() => {
+      response = await client.getPlugin(pluginId);
+      await response.delete();
+      if(response.data) {
         this.fetchPlugins();
-      });
+      }
```

else, maybe throw an error to trigger UI notification?

---

done